components/raftstore/src/store/fsm/apply.rs

```diff
@@ -697,6 +697,18 @@ where
     /// the source peer has applied its logs and pending entries
     /// are all handled.
     pending_msgs: Vec<Msg<E>>,
+    message_count: i64,
+}
+
+impl<E> Drop for YieldState<E>
+where
+    E: KvEngine,
+{
+    fn drop(&mut self) {
+        if self.message_count > 0 {
+            APPLY_PENDING_MSGS_COUNTER.inc_by(self.message_count);
```

Comment:
I think you can refer to how `APPLY_PENDING_BYTES_GAUGE` works. I think it's enough to mutate the counter in the same places.

Reply:
I don't understand what you mean, can you provide more details?

I think it's difficult mutate message counter and entries gauge in the same places.

Reply:
What I mean is counting the messages in the same place as `APPLY_PENDING_BYTES_GAUGE`. So when constructing `Apply`, besides updating pending bytes, it also updating pending entries:
```
APPLY_PENDING_BYTES_GAUGE.add(entries_mem_size);
APPLY_PENDING_ENTRY_COUNTER.add(entries.len());
```

Reply:
Fixed.  PTAL~
But for now, I am still confused about why the entry length is used for this metric.

Thanks for you reviewing.